### PR TITLE
solve issues 55;56;59;60;67...81;83

### PR DESCRIPTION
**Pull request by berndfinger**
#55: Use ansible reboot module
#56: Role is not recognizing cloud eus/e4s repositories
#59: feature idea: moving reboot to after all configuration changes
#60: handler: regenerate grub2 conf not working on ansible 2.8
#67: reboot fails on RHEL 8
#68: RHEL 8 PPCLE Packages
#69: Add support for --check (-c) option
#70: role fails, if firewalld is not installed
#71: update_only is not a valid parameter for package module
#72: required for update and reboot
#73: role internal variables should start with two underscores
#74: fix indentation
#75: Support for single step execution or checking of SAP notes
#76: no need to handle selinux in this role
#77: 2455582.yml can be removed.
#78: Support RHEL 7 and RHEL 8 and both architectures when checking enabled repos
#79: No longer support strict version checking for SAP HANA
#80: Add a warning to README.md
#81: Enable setting the RHEL release to a certain fixed minor release
#83: Warning needs to be rephrased.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,106 +47,121 @@ To get a personal developer edition of RHEL for SAP solutions, please register a
...
 You can use the [disk-init](https://galaxy.ansible.com/mk-ansible-roles/disk-init/) role to automate this process
 
-If you want to use this system in production make sure time service is configured correctly. You can use [rhel-system-roles](https://access.redhat.com/articles/3050101) to automate this
+If you want to use this system in production, make sure that the time service is configured correctly. You can use [rhel-system-roles](https://access.redhat.com/articles/3050101) to automate this.
+
+Note
+----
+For finding out which SAP notes will be used by this role, please check the contents of variable __sap_hana_preconfigure_sapnotes in files vars/*.yml (choose the file which matches your OS distribution and version)
+
+Please do not run this role against a productive SAP HANA system. The role will unconditionally make changes to the managed node, which might not be intended on productive SAP HANA systems.
```
Comment (richm) [inline README.md]: "production" not "productive"

File: tasks/RedHat/generic/configure-tuned.yml [hunk 2]
```diff
@@ -35,5 +35,52 @@
...
+      check_mode: no
+      register: current_profile_ppc64le
+      changed_when: false
+    - debug:
+        var: current_profile_ppc64le.stdout_lines, current_profile_ppc64le.stderr_lines
+
+    - name: On RHEL 8.0 on ppc64le, switch to tuned profiles sap-hana sap-hana-ppc64le if not currently active
+      block:
+        - name: Switch to tuned profiles sap-hana sap-hana-ppc64le if not currently active
+          shell: /usr/sbin/tuned-adm profile sap-hana sap-hana-ppc64le
```
Comment (richm) [inline tasks/RedHat/generic/configure-tuned.yml]: you can use `command` here instead of `shell`

**Review by richm — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (berndfinger) [reply README.md]: I rephrased the warning in https://github.com/berndfinger/sap-hana-preconfigure/commit/ce162cb8073e086ab19ffd37b3a808b070c7cfaf .

File: tasks/RedHat/generic/regenerate-grub2-conf.yml [hunk 3]
```diff
@@ -1,33 +1,30 @@
...
-  - name: "Run grub-mkconfig (BIOS mode)"
-    command: bash -lc "grub2-mkconfig -o /boot/grub2/grub.cfg"
-    register: command_result
-    when:
-      - not efi_exists.stat.exists
-      - sap_hana_preconfigure_run_grub2_mkconfig
-  - debug:
-      var: command_result.stdout_lines, command_result.stderr_lines
+- name: "Run grub-mkconfig (BIOS mode)"
+  command: bash -lc "grub2-mkconfig -o /boot/grub2/grub.cfg"
```
[the remark this answers is not in the document]
Comment (berndfinger) [reply tasks/RedHat/generic/regenerate-grub2-conf.yml]: command without bash should actually work fine. Will test and change if passed.

File: tasks/RedHat/generic/configure-tuned.yml (hunk 2 above)
Comment (berndfinger) [reply tasks/RedHat/generic/configure-tuned.yml]: +1 :-)

Comment (berndfinger) [reply tasks/RedHat/generic/configure-tuned.yml]: done - https://github.com/berndfinger/sap-hana-preconfigure/commit/318a2c6ec8d7a63efd457e7f2b39bd33326d41ab

File: tasks/RedHat/generic/regenerate-grub2-conf.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (berndfinger) [reply tasks/RedHat/generic/regenerate-grub2-conf.yml]: done - https://github.com/berndfinger/sap-hana-preconfigure/commit/318a2c6ec8d7a63efd457e7f2b39bd33326d41ab

**Review by richm — APPROVED** [review]
lgtm

**Review by rhmk — APPROVED** [review]
sounds goo to me

File: README.md [hunk 4]
```diff
@@ -118,26 +120,27 @@ sap_hana_preconfigure_fail_if_reboot_required
...
+If you do not want the role to switch to tuned profile sap-hana, set the following variable to `no`. Default is `yes`. In case of `yes`, variable `sap_hana_preconfigure_use_tuned_where_possible` (see below) should be set to `yes` as well.
 ```yaml
 sap_hana_preconfigure_switch_to_tuned_profile_sap_hana
 ```
 
 ### Use tuned profile sap-hana where possible
-If you do not want to use the tuned profile sap-hana for configuring kernel parameters (where possible), and have the role configure them instead, set the following variable to `no`. Default is `yes`.
+If you do not want to use the tuned profile sap-hana for configuring kernel parameters (where possible), and have the role configure them by changing the kernel command line instead, set the following variable to `no`. Default is `yes`. In case of `yes`, variable `sap_hana_preconfigure_switch_to_tuned_profile_sap_hana` (see above) should be set to `yes` as well.
+Note: If this variable is set to `yes`, the role will not modify GRUB_CMDLINE_LINUX in /etc/default/grub, no matter how `sap_hana_preconfigure_modify_grub_cmdline_linux` (see below) is set.
```
Comment (richm) [inline README.md:130]: @berndfinger not suggesting you do this now, but possibly for a future release - did you investigate using the tuned profile to configure the bootloader cmdline?

**Review by richm — APPROVED** [review]
lgtm

File: README.md (hunk 4 above)
Comment (berndfinger) [reply README.md:130]: Yes, I found that the three kernel boot cmdline parameters which are configured by this role are all covered by tuned-profiles-sap-hana. Here's the list:
CMDLINE: transparent_hugepage=never
tuned-profiles-sap-hana: transparent_hugepages=never
See https://github.com/berndfinger/sap-hana-preconfigure/blob/006a1c907953382860486528cdff13c344632309/tasks/RedHat/generic/disable-thp.yml#L6

CMDLINE: processor.max_cstate=1 intel_idle.max_cstate=1
tuned-profiles-sap-hana: force_latency=70 or force_latency=cstate.id:3|70
See https://github.com/berndfinger/sap-hana-preconfigure/blob/006a1c907953382860486528cdff13c344632309/tasks/RedHat/generic/configure-c-states-for-lower-latency.yml#L6

**Pull request merged**